atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraRequestExceptionHandler.java

```diff
@@ -98,13 +98,17 @@ RequestExceptionHandlerStrategy getStrategy() {
         if (ex instanceof TTransportException
                 && ex.getCause() != null
                 && (ex.getCause().getClass() == SocketException.class)) {
-            log.error(CONNECTION_FAILURE_MSG, numberOfAttempts, ex);
-            String errorMsg =
-                    MessageFormatter.format(CONNECTION_FAILURE_MSG, numberOfAttempts).getMessage();
+            log.warn(CONNECTION_FAILURE_MSG_WITH_EXCEPTION_INFO,
+                    SafeArg.of("numTries", numberOfAttempts),
+                    SafeArg.of("exceptionClass", ex.getClass().getTypeName()),
+                    UnsafeArg.of("exceptionMessage", ex.getMessage()));
+            String errorMsg = MessageFormatter.format(CONNECTION_FAILURE_MSG, numberOfAttempts).getMessage();
             throw (K) new TTransportException(((TTransportException) ex).getType(), errorMsg, ex);
         } else {
-            log.error("Tried to connect to cassandra {} times.",
-                    SafeArg.of("numTries", numberOfAttempts), ex);
+            log.warn("Tried to connect to cassandra {} times. Exception message was: {} : {}",
+                    SafeArg.of("numTries", numberOfAttempts),
+                    SafeArg.of("exceptionClass", ex.getClass().getTypeName()),
+                    UnsafeArg.of("exceptionMessage", ex.getMessage()));
```

Comment:
In general one advantage of logging the exception here is that you get the (safe) stack-trace, though I guess in these circumstances it isn't normally very useful?

Reply:
We also log on the first failure (at INFO), so are not losing this information assuming we keep failing for the same reason.